apps/designer/app/routes/rest/update.ts

```diff
@@ -0,0 +1,75 @@
+import type { ActionArgs } from "@remix-run/node";
+import { prisma } from "@webstudio-is/prisma-client";
+import { Styles } from "@webstudio-is/react-sdk";
+import { Messages, type StylesMessage } from "~/shared/stores";
+
+const loadStylesByTreeId = async (treeId: string) => {
+  const tree = await prisma.tree.findUnique({
+    where: { id: treeId },
+  });
+
+  if (tree === null) {
+    return [];
+  }
+
+  return Styles.parse(JSON.parse(tree.styles));
+};
+
+export const updateStyles = async (messages: StylesMessage[]) => {
```

Comment:
Just curious about compute time of all of this. Zod.parse messages and styles, iterate over mesage and find style.

Reply:
What with compute time? I run zod parse only once right after request.json().

Reply:
Ah, got it. You mean also parsing db data. This is done once too. And actual data is very simple comparing to denormalized tree. Should be faster.